crates/noirc_evaluator/src/ssa_refactor/ssa_builder/mod.rs

```diff
@@ -47,8 +52,9 @@ impl FunctionBuilder {
     /// A FunctionBuilder can always only work on one function at a time, so care
     /// should be taken not to finish a function that is still in progress by calling
     /// new_function before the current function is finished.
-    pub(crate) fn new_function(&mut self, name: String, function_id: FunctionId) {
-        let new_function = Function::new(name, function_id);
+    pub(crate) fn new_function(&mut self, name: String, function_id: FunctionId, is_unsafe: bool) {
+        let runtime = if is_unsafe { RuntimeType::Unsafe } else { RuntimeType::Normal };
+        let new_function = Function::new(name, function_id, runtime);
```

Comment:
It is a bit odd this function takes `is_unsafe: bool` while `new` takes `runtime: RuntimeType` directly. I think we should do the same thing with Functions and remove the argument from the constructor and add a mutator method for the function.

Reply:
I'd be in favor of having a `new_brillig_function` method

Reply:
Pushed a fix to use `new_function` and `new_brillig_function`